### PR TITLE
Fix iOS compilation issues

### PR DESCRIPTION
**Pull request by pandamicro**

### CODE REVIEW

**Comment by pandamicro** [comment]
https://github.com/cocos-creator/3d-tasks/issues/1838

File: templates/js-template-default/frameworks/runtime-src/proj.ios_mac/HelloJavascript.xcodeproj/project.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+	<key>IDEDidComputeMac32BitWarning</key>
+	<true/>
+</dict>
+</plist>
```
Comment (minggo) [inline templates/js-template-default/frameworks/runtime-src/proj.ios_mac/HelloJavascript.xcodeproj/project.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist]: 这个文件是干什么用的？

Comment (pandamicro) [reply templates/js-template-default/frameworks/runtime-src/proj.ios_mac/HelloJavascript.xcodeproj/project.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist]: 不知道，我用 xcode 打开，必然创建这个文件

Comment (pandamicro) [reply templates/js-template-default/frameworks/runtime-src/proj.ios_mac/HelloJavascript.xcodeproj/project.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist]: Xcode 9.3 adds a new IDEWorkspaceChecks.plist file to a workspace's shared data, to store the state of necessary workspace checks. Committing this file to source control will prevent unnecessary rerunning of those checks for each user opening the workspace. (37293167)

File: cocos/renderer/gfx-gles3/GLES3GPUObjects.h [hunk 2]
```diff
@@ -121,6 +121,12 @@ struct GLES3GPUUniformSampler {
 typedef vector<GLES3GPUUniformSampler>::type GLES3GPUUniformSamplerList;
 
 struct GLES3GPUShaderStage {
+  GLES3GPUShaderStage(GFXShaderType type, String source, GFXShaderMacroList macros, GLuint gl_shader = 0)
```
Comment (minggo) [inline cocos/renderer/gfx-gles3/GLES3GPUObjects.h]: 传进来的参数名字和类成员变量名字最好不一样，比如 _type

Comment (pandamicro) [reply cocos/renderer/gfx-gles3/GLES3GPUObjects.h]: 改了

**Pull request merged**